### PR TITLE
Adding new module gfaffix

### LINKED ISSUE #1134: new module: gfaffix
I think it would be good to have a module for [gfaffix](https://bioconda.github.io/recipes/gfaffix/README.html)

- [x] This module does not exist yet with the [`nf-core modules list`](https://github.com/nf-core/tools#list-modules) command
- [x] There is no [open pull request](https://github.com/nf-core/modules/pulls) for this module
- [x] There is no [open issue](https://github.com/nf-core/modules/issues) for this module
- [x] If I'm planning to work on this module, I added myself to the `Assignees` to facilitate tracking who is working on the module

https://github.com/nf-core/pangenome/issues/70

### PR DESCRIPTION
**Pull request by heuermh**
Fixes #1134 

## PR checklist

Closes #XXX 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
    - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd`

### CODE REVIEW

**Comment by heuermh** [comment]
May have missed something, the CI test did not run

**Comment by heuermh** [comment]
That's better 😄

**Comment by subwaystation** [comment]
Let's wait for https://github.com/nf-core/test-datasets/pull/406, so we can use the proper test data.

**Comment by heuermh** [comment]
This one runs fine and produces non-empty output on the sars-cov-2 `assembly.gfa`, we could merge as-is.

File: modules/gfaffix/main.nf
Comment (SPPearce) [inline modules/gfaffix/main.nf:3]: Does this need `process_medium`? It is is only currently using one core, does it need more than 6Gb RAM?

**Review by SPPearce — APPROVED** [review]
This looks fine as is. It looks like the test-data was merged, if you want to add tests for that too.

File: modules/gfaffix/main.nf
Comment (subwaystation) [reply modules/gfaffix/main.nf:3]: Dependent on the input graph it might need over 100Gb RAM. But it only uses 1 thread.

File: modules/gfaffix/main.nf
Comment (subwaystation) [inline modules/gfaffix/main.nf:8]: The most recent version is 0.1.4 https://anaconda.org/bioconda/gfaffix.

Comment (subwaystation) [reply modules/gfaffix/main.nf:8]: So before merging I would suggest to update.

File: modules/gfaffix/main.nf
Comment (subwaystation) [reply modules/gfaffix/main.nf:3]: But for small graphs 1Gb suffices. It highly depends on the pangenome graph.

Comment (SPPearce) [reply modules/gfaffix/main.nf:3]: process_medium is allocated 36Gb by default on the first attempt. Is that likely to work on "typical" input, or would it need more?

File: modules/gfaffix/main.nf
Comment (SPPearce) [reply modules/gfaffix/main.nf:8]: I would agree, would be good to bump the version before we merge.

File: modules/gfaffix/main.nf
Comment (subwaystation) [reply modules/gfaffix/main.nf:3]: I am sorry to be so vague, but it really depends on the input. We can build pangenome graphs from HLA genes https://github.com/pangenome/pggb/tree/master/data/HLA [1Gb], from 7 yeast assemblies https://pggb.readthedocs.io/en/latest/rst/tutorials/sequence_partitioning.html#steps [10-20Gb], or from 94 human haplotypes https://github.com/human-pangenomics/hpp_pangenome_resources [~150-200Gb]. More complex plant genomes or more number of haplotypes would require more RAM. 

There is no typical input here. Maybe later there is a way to estimate the RAM requirements before running the process, but that's not possible at the moment.

**Comment by heuermh** [comment]
I'm not sure what this lint error is referring to
```
╭─ [✗] 1 Module Test Failed ───────────────────────────────────────────────────╮
│                                           ╷                ╷                 │
│ Module name                               │ File path      │ Test message    │
│╶──────────────────────────────────────────┼────────────────┼────────────────╴│
│ gfaffix                                   │ modules/gfaff… │ when: condition │
│                                           │                │ has been        │
│                                           │                │ removed         │
│                                           ╵                ╵                 │
╰──────────────────────────────────────────────────────────────────────────────╯
```

**Comment by subwaystation** [comment]
> 

@FriederikeHanssen @mirpedrol  Can you help us here please?

**Comment by subwaystation** [comment]
@heuermh The md5sum is unhappy. https://github.com/nf-core/modules/actions/runs/3136282988/jobs/5093001920. I suspect that's because the new version outputs a different file than the one you worked with in January.

**Comment by heuermh** [comment]
> md5sum is unhappy

I suppose that is to be expected, and can be updated to match the new output.  I'm not sure what the nf-core linter is complaining about, I can't run it locally because of M1 Docker/conda issues.

**Comment by FriederikeHanssen** [comment]
> I'm not sure what this lint error is referring to
> 
> ```
> ╭─ [✗] 1 Module Test Failed ───────────────────────────────────────────────────╮
> │                                           ╷                ╷                 │
> │ Module name                               │ File path      │ Test message    │
> │╶──────────────────────────────────────────┼────────────────┼────────────────╴│
> │ gfaffix                                   │ modules/gfaff… │ when: condition │
> │                                           │                │ has been        │
> │                                           │                │ removed         │
> │                                           ╵                ╵                 │
> ╰──────────────────────────────────────────────────────────────────────────────╯
> ```

you are missing the `when` section:
```
    when:
    task.ext.when == null || task.ext.when

```

should be added in please :)

File: modules/gfaffix/main.nf
Comment (FriederikeHanssen) [reply modules/gfaffix/main.nf:3]: It would be good to be fairly conservative, the resource requests can always be increased in the pipelines. if it only uses one thread, I would suggest process_single

**Comment by heuermh** [comment]
Thank you, @FriederikeHanssen!

File: modules/gfaffix/main.nf
Comment (heuermh) [reply modules/gfaffix/main.nf]: Resolved by commit fdeeb2bc6a4f79bb6fc69a7d4e4554f04802753c

**Review by SPPearce — APPROVED** [review]
Happy to merge once we figure out the best process option.

File: modules/gfaffix/main.nf
Comment (heuermh) [reply modules/gfaffix/main.nf:3]: I'm ok defaulting to `process_single`

Comment (subwaystation) [reply modules/gfaffix/main.nf:3]: Cool! Once this is in, we can merge :)

**Review by subwaystation — APPROVED** [review]
Great job @heuermh!
Only the label `process_single` is missing I think.

**Review by SPPearce — APPROVED** [review]

**Comment by subwaystation** [comment]
Thank you very much for this @heuermh!

**Comment by heuermh** [comment]
Thank you, @subwaystation @SPPearce

**Pull request merged**
